web/app.js

```diff
@@ -878,24 +895,16 @@ const PDFViewerApplication = {
       typeof PDFJSDev === "undefined" ||
       PDFJSDev.test("MOZCENTRAL || GENERIC")
     ) {
-      if (featureId) {
-        this._recordFallbackErrorTelemetry(featureId);
-      }
-
-      // For PDFs that contain script and form errors, we should only trigger
-      // the fallback once the user has interacted with the page.
-      if (this._delayedFallbackFeatureIds.length >= 1 && this._hasInteracted) {
-        featureId = this._delayedFallbackFeatureIds[0];
-        // Reset to prevent all click events from showing fallback bar.
-        this._delayedFallbackFeatureIds = [];
-      }
+      this.externalServices.reportTelemetry({
```

Comment:
Aren't forms and JavaScript now reported twice? They trigger `_delayedFallback` which unconditionally reports the telemetry and creates a callback that calls `fallback` which also reports the telemetry here.

Reply:
Yes, but this isn't a problem neither here nor generally since the mozilla-central side already deals with duplicates; please see https://searchfox.org/mozilla-central/rev/e11d919cdc8a909354eb2c3e19904d9229c55d88/toolkit/components/pdfjs/content/PdfStreamConverter.jsm#433-499